seedVault/index.ts

```diff
@@ -0,0 +1,94 @@
+export interface StorageAdapter {
+  get(key: string): Promise<string> | string | null;
+  set(key: string, value: string): Promise<void> | void;
+}
+
+export interface CryptoUtilsAdapter {
+  encrypt(data: string, password: string): Promise<string>;
+  decrypt(data: string, password: string): Promise<string>;
+  hash(data: string, salt: string): Promise<string>;
+  generateRandomBytes(length: number): string;
+}
+
+export interface SeedVaultConfig {
```

Comment:
These are probably better off as types instead of interfaces. Also, in `CryptoUtilsAdapter`, could we make explicit types for the encrypt and decrypt functions and assign those types to the related functions in the encryptionUtils file? That way there is a clear link between them.